python/cudf/cudf/core/_base_index.py

```diff
@@ -162,6 +167,38 @@ def _clean_nulls_from_index(self):
         else:
             return self
 
+    @property
+    def is_monotonic(self):
+        """Return boolean if values in the object are monotonic_increasing.
+
+        This property is an alias for :attr:`is_monotonic_increasing`.
+
+        Returns
+        -------
+        bool
+        """
+        return self.is_monotonic_increasing
+
+    @property
+    def is_monotonic_increasing(self):
```

Comment:
Why can we use `self.is_monotonic_increasing` above but then here we're raising a not implemented error?

Reply:
We want `is_monotonic` to be an alias for `is_monotonic_increasing` for all types of indexes. The problem is that the implementation of `is_monotonic_increasing` is different for different types. By using `is_monotonic_increasing` above but making it raise here, we ensure that for any index that defines `is_monotonic_increasing` we will observe the correct behavior for `is_monotonic` without duplicating the logic, but we will get an error for subclasses that don't override this method.